src/Microsoft.AspNetCore.Server.Kestrel.Core/KestrelServerLimits.cs

```diff
@@ -170,13 +152,51 @@ public TimeSpan KeepAliveTimeout
         /// </remarks>
         public TimeSpan RequestHeadersTimeout
         {
-            get
+            get => _requestHeadersTimeout;
+            set => _requestHeadersTimeout = value;
+        }
+
+        /// <summary>
+        /// Gets or sets the maximum number of connections that will accept requests. When set to null, the number of connections is unlimited.
+        /// </summary>
+        /// <remarks>
+        /// Defaults to <see cref="uint.MaxValue"/>.
+        /// </remarks>
+        public uint? MaxConcurrentConnections
+        {
+            get => _maxConcurrentConnections;
+            set
             {
-                return _requestHeadersTimeout;
+                if (value.HasValue && value <= 0)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value), CoreStrings.PositiveIntRequired);
+                }
+                _maxConcurrentConnections = value;
             }
+        }
+
+        /// <summary>
+        /// Gets or sets the maximum number of upgraded connections that can be open. When set to null, the number of upgraded connections is unlimited.
+        /// An upgraded connection is one that has been switched from HTTP to another protocol, such as WebSockets.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// Defaults to <see cref="uint.MaxValue"/>.
+        /// </para>
+        /// <para>
+        /// This limit is independent of the value of <see cref="MaxConcurrentConnections" />.
```

Comment:
A clearer way to phrase this might be `Upgraded connections do not count towards the <see cref="MaxConcurrentConnections" /> limit.`